code/game/objects/structures/gas_stand.dm

```diff
@@ -0,0 +1,240 @@
+		else if(istype(tank,/obj/item/weapon/tank/oxygen))
+			overlays += "tank_oxyg"
+		//else if(istype(tank,/obj/item/weapon/tank/phoron))
+		//	overlays += "tank_phoron"
+		//else if(istype(tank,/obj/item/weapon/tank/hydrogen))
+		//	overlays += "tank_hydro"
+		else
+			overlays += "tank_other"
+
+/obj/structure/gas_stand/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	if(breather)
+		breather.internal = null
+		if(internalsHud)
+			internalsHud.icon_state = "internal0"
+	if(tank)
+		qdel(tank)
+	if(breather)
+		breather.remove_from_mob(contained)
+		src.visible_message("<span class='notice'>The mask rapidly retracts just before /the [src] is destroyed!</span>")
+	qdel(contained)
+	contained = null
+	breather = null
+	return ..()
+
+/obj/structure/gas_stand/attack_robot(var/mob/user)
+	if(Adjacent(user))
+		attack_hand(user)
+
+/obj/structure/gas_stand/MouseDrop(var/mob/living/carbon/human/target, src_location, over_location)
+	..()
+	if(istype(target))
+		if(!can_apply_to_target(target, usr)) // There is no point in attempting to apply a mask if it's impossible.
+			return
+		usr.visible_message("\The [usr] begins placing the mask onto [target]..")
+		if(!do_mob(usr, target, 25) || !can_apply_to_target(target, usr))
+			return
+		// place mask and add fingerprints
+		usr.visible_message("\The [usr] has placed \the mask on [target]'s mouth.")
+		attach_mask(target)
+		src.add_fingerprint(usr)
+		update_icon()
+		START_PROCESSING(SSobj,src)
+
+/obj/structure/gas_stand/attack_hand(mob/user as mob)
+	if (tank && is_loosen)
+		user.visible_message("<span class='notice'>\The [user] removes \the [tank] from \the [src].</span>", "<span class='notice'>You remove \the [tank] from \the [src].</span>")
+		user.put_in_hands(tank)
+		src.add_fingerprint(user)
+		tank.add_fingerprint(user)
+		tank = null
+		update_icon()
+		return
+	if (!tank)
+		to_chat(user, "<span class='warning'>There is no tank in \the [src]!</span>")
+		return
+	else
+		if (valve_opened)
+			src.visible_message("<span class='notice'>\The [user] closes valve on \the [src]!</span>")
+			if(breather)
+				if(internalsHud)
+					internalsHud.icon_state = "internal0"
+				breather.internal = null
+			valve_opened = FALSE
+			update_icon()
+		else
+			src.visible_message("<span class='notice'>\The [user] opens valve on \the [src]!</span>")
+			if(breather)
+				breather.internal = tank
+				if(internalsHud)
+					internalsHud.icon_state = "internal1"
+			valve_opened = TRUE	
+			playsound(get_turf(src), 'sound/effects/internals.ogg', 100, 1)
+			update_icon()
+			START_PROCESSING(SSobj,src)
+
+/obj/structure/gas_stand/proc/attach_mask(var/mob/living/carbon/C)
+	if(C && istype(C))
+		contained.forceMove(get_turf(C))
+		C.equip_to_slot(contained, slot_wear_mask)
+		if(tank)
+			tank.forceMove(C)
+		breather = C
+		if(breather.HUDneed.Find("internal"))
+			internalsHud = breather.HUDneed["internal"]
+
+/obj/structure/gas_stand/proc/can_apply_to_target(var/mob/living/carbon/human/target, mob/user as mob)
```

Comment:
`user` can be not human, not Adjacent or even dead.

Reply:
added human check, adjacent check is present and there should be no difference if mob is dead or not you can still place mask on a dead body

Reply:
@LagannPM 
User also should be adjacnt, not only target. Same for dead.